### PR TITLE
receiver: use None for serial number of c534

### LINKED ISSUE #636: c534 receivers don't have serial numbers and only can pair two devices
c534 are limited receivers.  They don't have serial numbers and can only pair a limited number of devices.

There is special code to handle them that gives them a serial number of 0 (the integer) and pairing number of 6.  Both cause problems.

idefix testing> ../bin/solaar -D /dev/hidraw10 show 99
solaar: error: 'int' object has no attribute 'lower'

This happens because 0 is not a string.

Setting the pairing number to 6 give a very false impression of the receiver.

A fix is in the works.

### PR DESCRIPTION
**Pull request by pfps**
This only fixes the error caused by using an integer for the serial number of c534.

It does not fix the number of pairable devices.  That needs more of my earlier changes for the c534, which are waiting for information from Logitech.

Addresses #636

### CODE REVIEW

**Comment by FFY00** [comment]
I think it should be `None`.

**Comment by pfps** [comment]
That might be best but there is at least one place that assumes that lower can be done on the serial number.  I'll see if I can easily find every place the serial number is used.

**Comment by pfps** [comment]
OK, it looks as if there is only one place that *needs* to be changed.  Of course None now shows up as the serial number of these receivers in the UI.  I actually think this makes sense, as they don't have serial numbers.  It's also better than not showing the serial number at all, I think, as otherwise there are likely to be questions about why the serial number isn't showing up.

**Comment by pfps** [comment]
Yes, I have several c534 receivers available as well as a Unifying receiver.

File: lib/solaar/cli/__init__.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def _find_receiver(receivers, name):
 	assert name
 
 	for r in receivers:
-		if name in r.name.lower() or name == r.serial.lower():
+		if name in r.name.lower() or ( (name == r.serial.lower()) if (r.serial is not None) else False) :
```
Comment (FFY00) [inline lib/solaar/cli/__init__.py]:
```python
if name in r.name.lower() or (r.serial and name == r.serial.lower()):
```

This should work but please test.

Comment (pfps) [reply lib/solaar/cli/__init__.py]: I'm going to use is not None to better handle the edge case of an empty string for a serial number.

Comment (pfps) [reply lib/solaar/cli/__init__.py]: OK, I amended and force-pushed the change so that there is only one commit.

Comment (FFY00) [reply lib/solaar/cli/__init__.py]: An empty string will also evaluate as false. I'd prefer `r.serial` instead of `r.serial is None` as that is simpler but I'll accept either way.

**Comment by pfps** [comment]
@Lekensteyn Yes, but there are two aspects to #636 and this only fixes one of them.

File: lib/solaar/cli/__init__.py (hunk 1 above)
Comment (pfps) [reply lib/solaar/cli/__init__.py]: I actually used r.serial is None so that falsy serial numbers can be checked.  I don't expect it to ever happen, but I didn't want the code to be tripped up in this vanishingly unlikely situation.

**Comment by pfps** [comment]
@FFY00 What is required to have this PR merged?
